python/scannerpy/database.py

```diff
@@ -318,15 +318,13 @@ def start_cluster(self, master, workers):
             self._master_conn = None
             self._worker_conns = None
             machine_params = self._bindings.default_machine_params()
-            res = self._bindings.start_master(self._db,
-                                              self.config.master_port).success
-            assert res
+            assert self._bindings.start_master(
+                self._db, self.config.master_port).success
+            assert self._connect_to_master()
```

Comment:
It's possible for statements guarded by asserts to be conditioned out in Python. https://docs.python.org/3/reference/simple_stmts.html#assert

We should assert on the result value separately (or have a proper error message).